src/main/java/com/google/neighborgood/servlets/EditTaskServlet.java

```diff
@@ -20,27 +20,94 @@
 import com.google.appengine.api.datastore.EntityNotFoundException;
 import com.google.appengine.api.datastore.Key;
 import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Transaction;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.neighborgood.helper.RewardingPoints;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that handles the request for editing the details and rewards for a certain task. */
+/** Servlet that handles the request for editing tasks. */
 @WebServlet("/tasks/edit")
 public class EditTaskServlet extends HttpServlet {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
     String keyString = request.getParameter("task-id");
-    int rewardPts = getRewardingPoints(request, "reward-input");
-    if (rewardPts == -1) {
+    Key taskKey = KeyFactory.stringToKey(keyString);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity task;
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      System.err.println("User must be logged in to edit a task");
+      response.sendError(
+          HttpServletResponse.SC_UNAUTHORIZED,
+          "You must be logged in to perform this action on a task");
+    }
+
+    // Edits tasks that have been claimed by setting the "helper" property to the userId
+    // of the helper and changing the task's status to "IN PROGRESS"
+    if (request.getParameterMap().containsKey("action")) {
+      if (request.getParameter("action").equals("helpout")) {
```

Comment:
Can we join these two if statements?

if (request.getParameterMap().containsKey("action") && request.getParameter("action").equals("helpout")) {
  ...
}

Reply:
You’re right! I was afraid the second statement would be evaluated even if the first evaluated to false, but I just learnt the double ampersand follows short-circuiting Logic. I’ll change this, thanks for the suggestion!